api/v1alpha1/runner_types.go

```diff
@@ -31,7 +31,7 @@ type RunnerSpec struct {
 	Image string `json:"image"`
 
 	// +optional
-	Env []corev1.EnvVar `json:"env,omitempty"`
+	PodTemplate corev1.PodTemplate `json:"podTemplate,omitempty"`
```

Comment:
Could you add a `PodTemplate` field instead of replacing `Env` to keep your custom resources backward compatible?